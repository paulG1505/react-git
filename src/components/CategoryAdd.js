import React,{useState} from 'react'
import PropTypes from 'prop-types'

export const CategoryAdd = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('')
    const handleInput=({target})=>{
        //en el target esta el valor
        setInputValue(target.value);
    }
    const handleSubmit=(e)=>{
        //prevenir el volver a cargar la pagina
        e.preventDefault()
        //para que no envie vacio
        if(inputValue.trim().length>1){

            //copia de lo que tiene el arreglo e incrementa primero
            setCategories(cat=>[inputValue,...cat])
            setInputValue('')
        }

    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={inputValue}
                onChange={handleInput}
            />
        </form>
    )
}
//para que sea obigatorio el enviar
CategoryAdd.propTypes = {
    setCategories: PropTypes.func.isRequired
}
