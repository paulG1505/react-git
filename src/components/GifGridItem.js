import React from 'react'
import PropTypes from 'prop-types'
//impt snnipet

export const GifGridItem = ({url,title}) => {
    return (
        <div className="card animate__animated animate__backInRight">
            <img
                src={url} height='120px'
                // titulo alternativo
                alt={title}
            />
            <br/>
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes={
    title:PropTypes.string.isRequired,
    url:PropTypes.string.isRequired
}