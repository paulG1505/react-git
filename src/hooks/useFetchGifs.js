import { useState,useEffect } from "react"
import { getGifs } from "../helpers/getGifs"

export const useFetchGifs = (category) => {
    const [data, setData] = useState([])
    const [loading, setloading] = useState(true)
    useEffect(() => {
        getGifs(category)
            .then(setData)
        setloading(false)
        console.log(loading);
    }, [category])//si la categoria cambia que vuelva a ejecuatar
    return {loading,data};
}
