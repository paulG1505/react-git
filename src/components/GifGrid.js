import React from 'react'
import { GifGridItem } from './GifGridItem'
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    const { loading, data } = useFetchGifs(category);
    return (
        <>
            <h3 className="animate__animated animate__fadeInDown" >{category}</h3>
            {loading ? 'Cargando...' :

                <div className="card-grid">

                    {/* al objeto lo descomponemos */}
                    {
                        data.map((img) => (
                            <GifGridItem
                                {...img}
                                key={img.id}
                            />
                        ))
                    }
                </div>
            }
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,

}