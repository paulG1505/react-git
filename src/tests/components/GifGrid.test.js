const { shallow } = require("enzyme")
import '@testing-library/jest-dom'
import React from 'react'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas a GifGrid', () => {


    test('Comprobacion snapshot', () => {
        //simulando hook
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category='Hola' />)
        expect(wrapper).toMatchSnapshot();
    })
    test('Debe mostrar items cuando llegan las imagenes del useFetch', () => {
        //se debe simular comportamiento del hook por lo que se usa mock
        useFetchGifs.mockReturnValue({
            data: [{ id: '1', title: 'ho', url: 'https://' }],
            loading: false
        })
        
        const wrapper = shallow(<GifGrid category='Hola' />)
        //expect(wrapper).toMatchSnapshot();//evaluamos frente a la fotografia
        //evaluamos si no existe el loading
        expect(wrapper.find('p').exists()).toBe(false)
        //evaluamos que exista el grid item de acuerdo a los gifs que tenemos
        expect(wrapper.find('GifGridItem').length).toBe(1);
    })

})
