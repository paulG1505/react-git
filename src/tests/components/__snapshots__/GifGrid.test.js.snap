// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a GifGrid Comprobacion snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    Hola
  </h3>
  Cargando...
</Fragment>
`;
